### PR TITLE
esp32/boards: Added LOLIN_S2_MINI ESP32-S2 Board

### PR DESCRIPTION
**Pull request by sbonnick**
To support Lolin S2 Mini ESP32-S2 Variant board

LOLIN_S2_MINI Variant board leverages the ESP32-S2 chip. Equiped with onboard LED and button sensors. More information can be found here https://www.wemos.cc/en/latest/s2/s2_mini.html

### CODE REVIEW

**Comment by sbonnick** [comment]
FYI @mcauser

**Comment by sbonnick** [comment]
I have reduced the scope as well to jsut the S2_MINI board and not the pico. there was some additional feedback for that board however I dont own that and cannot test on it. So I will leave that to the Wemos to commit to after deciding how they want to handle the Network debug methods.

File: ports/esp32/boards/LOLIN_S2_MINI/modules/s2mini.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# LOLIN S2 MINI MicroPython Helper Library
+
+
+from micropython import const
+from machine import Pin, SPI, ADC
+import machine, time
```
Comment (dpgeorge) [inline ports/esp32/boards/LOLIN_S2_MINI/modules/s2mini.py:6]: unless there's a reason to include them, I'd suggest removing imports that are unused: SPI, ADC, machine, time

**Comment by dpgeorge** [comment]
Thanks for updating, merged in 0d9429f44cd23f5cfdd80d53adccbf384e23665a

**Pull request closed without merging**